GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -956,10 +959,34 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            if (_branchNameLabelOnClick != null)
+            {
+                remoteNameLabel.Click -= _branchNameLabelOnClick;
+            }
+
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            if (currentBranch == null)
+            {
+                branchNameLabel.Text = currentBranchName;
+                remoteNameLabel.Text = string.Empty;
+                return;
+            }
+
+            var pushTo = string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith)
+                ? Module.GetRemotes().Any(r => r.Name == "origin") ? $"(default) origin/{currentBranchName}" : _statusBarBranchWithoutRemote.Text
```

Comment:
```suggestion
                ? Module.GetRemotes().Any(r => r.Name == "origin") ? $"origin/{currentBranchName} (untracked)" : _statusBarBranchWithoutRemote.Text
```
thoughts?

Reply:
Yes better. But the question was also if this text should be in the link.
Or displayed outside which is a more intrusive change and required a new text label...

Reply:
I don't particularly mind it all being a link